rspack cdn example update

ScriptedAlchemy opened the pull request.

ScriptedAlchemy committed:
Update "@rspack/core" and "@rspack/dev-server" versions

Updated the versions of "@rspack/core" and "@rspack/dev-server" in several package.json files. The new version improves performance and provides additional features. These updates keep the project up-to-date with the latest development practices.

ScriptedAlchemy committed:
Merge branch 'master' into ts-example-rspack

ScriptedAlchemy committed:
update rspack

ScriptedAlchemy committed:
Add local webpack and setup rspack for development in NextJs and React apps

This commit introduces rspack in the build and dev scripts of NextJs and React apps, replacing webpack-cli. We've added the NEXT_PRIVATE_LOCAL_WEBPACK environment variable in our build and dev scripts for enabling local webpack. In the React app, a new rspack.config.js file has been created for configuring the newly introduced rspack.

ScriptedAlchemy committed:
Merge branch 'master' into ts-example-rspack

ScriptedAlchemy committed:
locks

renovate[bot] committed:
chore(deps): update dependency eslint-plugin-import to v2.29.1

ScriptedAlchemy committed:
Add resolve extensions in rspack.config.js

The update enables resolution of module paths for file types specified, which includes ".tsx", ".ts", ".jsx", ".js", ".json". This change will improve module dependency resolution by specifying the permissible file type extensions.

ScriptedAlchemy committed:
Add resolve extensions to webpack configuration

Added ".tsx", ".ts", ".jsx", ".js", and ".json" extensions in the resolve array of the rspack.config.js in both app1 and app2. These additional extensions will aid in the resolution of module paths, consequently enhancing module dependency management.

ScriptedAlchemy committed:
Update webpack and rspack configurations

Modified webpack.config.js in app-02 by removing some unnecessary spaces and included ".jsx", ".js", ".json", and ".mjs" extensions in the resolve configurations of the rspack.config.js in app-01, app-02 and app-03. This is to enhance the resolution of module paths and to improve module dependency management.

renovate[bot] committed:
chore(deps): update dependency less-loader to v11.1.4

ScriptedAlchemy committed:
Update .gitignore with new paths

Added new entries to .gitignore for JS and CSS files in the comprehensive-demo-react16 public folder. This change prevents unneeded static build files from being tracked and committed to the repository.

ScriptedAlchemy committed:
Update .gitignore with new paths

Added new entries to .gitignore for JS and CSS files in the comprehensive-demo-react16 public folder. This change prevents unneeded static build files from being tracked and committed to the repository.

ScriptedAlchemy committed:
Merge branch 'master' into rspack-examples

# Conflicts:
#	angular14-react/react-user-list/package.json
#	federated-css/expose-remotes/expose-tailwind-css-module/package.json
#	pnpm-lock.yaml

ScriptedAlchemy committed:
Update consumers-react compatibility status

Changed the status of compatibility for consumers-react with rspack from not compatible to compatible. This update in the documentation reflects the recent updates and improvements made in the compatibility of the consumers-react with rspack.

ScriptedAlchemy committed:
Update consumers-react compatibility status

Changed the status of compatibility for consumers-react with rspack from not compatible to compatible. This update in the documentation reflects the recent updates and improvements made in the compatibility of the consumers-react with rspack.

renovate[bot] committed:
chore(deps): update dependency postcss to v8.4.33

ScriptedAlchemy committed:
Merge branch 'master' into rspack-examples

# Conflicts:
#	federated-css/consumers-react/any-combination/rspack.config.js
#	output.md
#	pnpm-lock.yaml

ScriptedAlchemy committed:
locks

ScriptedAlchemy committed:
Merge remote-tracking branch 'origin/renovate/postcss-8.x' into rspack-examples

# Conflicts:
#	pnpm-lock.yaml

ScriptedAlchemy committed:
Merge remote-tracking branch 'origin/renovate/eslint-plugin-import-2.x' into rspack-examples

ScriptedAlchemy committed:
Merge remote-tracking branch 'origin/renovate/less-loader-11.x' into rspack-examples

ScriptedAlchemy committed:
locks

ScriptedAlchemy committed:
Merge branch 'master' into rspack-examples

ScriptedAlchemy committed:
cdn example

ScriptedAlchemy committed:
cdn example

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/module-federation/module-federation-examples?pullRequest=3521) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/module-federation/module-federation-examples?pullRequest=3521) before we can accept your contribution.<br/>**0** out of **2** committers have signed the CLA.<br/><br/>:x: ScriptedAlchemy<br/>:x: renovate[bot]<br/><hr/>**ScriptedAlchemy** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/module-federation/module-federation-examples?pullRequest=3521) it.</sub>

ScriptedAlchemy committed:
Update scripts and dependencies in package.json files

In this commit, the scripts and dependencies within the package.json files of multiple applications and the remote library have been updated. The Webpack commands were replaced by rspack for the 'start' and 'build' scripts, while preserving the former as legacy options. New rspack related devDependencies were also added. The scripts on the root level package.json were updated to use pnpm, optimizing for a monorepo setup. Finally, the 'dev' script in the remoteLibrary was renamed to 'start'.

ScriptedAlchemy committed:
Change devServer port in rspack.dev.js

In this commit, the port setting for the development server in the rspack.dev.js file of the remoteLibrary has been updated. Previously set to 3002, the port number is now changed to 3003 to potentially avoid port collision and provide smooth running of the development server.
